### PR TITLE
Implement effect of duel card #45

### PR DESCRIPTION
**Pull request by IvanovAndrew**

### CODE REVIEW

File: Bang/GameEvents/CardEffects/DuelActionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        public override HandlerState ApplyEffect(Game.Gameplay gameplay, Player victim, BangGameCard card)
+        {
+            DefenceAgainstDuel response = new DefenceAgainstDuel {Player = victim};
+            
+            return new WaitingBangAfterDuelState(gameplay, victim, gameplay.PlayerTurn){SideEffect = response};
+        }
+    }
+
+    internal class WaitingBangAfterDuelState : HandlerState
```
Comment (bashnia007) [inline Bang/GameEvents/CardEffects/DuelActionHandler.cs:19]: We started to make State classes as separate files. Maybe we should continue this practice? Or change previous files to this template

File: Bang.Tests/DuelSpecification.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+            // Act
+            victim.Defense(bangCard);
+            
+            // Assert
+            victim.Hand.Should().NotContain(bangCard);
+        }
+
+        [Fact]
+        public void If_victim_and_attacker_play_by_one_bang_card_then_victim_will_lose_one_life()
```
Comment (bashnia007) [inline Bang.Tests/DuelSpecification.cs:115]: It seems the test name doesn't reflect correctly the logic. Here attacker plays 2 bang cards

Comment (IvanovAndrew) [reply Bang.Tests/DuelSpecification.cs:115]: I don't see that. Attacker player plays `Duel` and `Bang`
```
actor.PlayCard(duelCard, victim);
victim.Defense(anotherBangCard);
actor.Defense(bangCard);
```

File: Bang/GameEvents/CardEffects/DuelActionHandler.cs (hunk 1 above)
Comment (IvanovAndrew) [reply Bang/GameEvents/CardEffects/DuelActionHandler.cs]: Fixed

**Pull request merged**